nifi-nar-bundles/nifi-standard-bundle/nifi-standard-reporting-tasks/src/main/java/org/apache/nifi/controller/MonitorMemory.java

```diff
@@ -218,7 +218,11 @@ public void onTrigger(final ReportingContext context) {
         }
 
         final double percentageUsed = (double) usage.getUsed() / (double) usage.getMax() * 100D;
-        if (bean.isCollectionUsageThresholdSupported() && bean.isCollectionUsageThresholdExceeded()) {
+        final String percentageThreshold = threshold.substring(0, threshold.length() - 1);
+        final double usageThreshold = Double.parseDouble(percentageThreshold);
+        // In certain scenarios in the monitored memory bean the gcSensor can get stuck in 'on' state before the usage would reach the threshold
+        // and this will cause false exceeded state until the next garbage collection. To eliminate this we are adding a condition with the calculated usage threshold.
+        if (bean.isCollectionUsageThresholdSupported() && bean.isCollectionUsageThresholdExceeded() && percentageUsed > usageThreshold) {
```

Comment:
I don't think we need check for bean.isCollectionUsageThresholdExceeded() anymore

Reply:
I wanted to keep the gcSensor and MemoryPoolMXBean side calculation and additionally calculate the threshold on NiFi side to make sure that the warning message is raised if both side calculation is the same. What do you think about this?

Reply:
So if I understand you correctly, you are leaving this in intentionally, as further protection for the false alert.
That sounds good to me.

Reply:
Yes that's what I meant.